src/networking.c

```diff
@@ -279,23 +280,20 @@ int prepareClientToWrite(client *c) {
  * -------------------------------------------------------------------------- */
 
 /* Attempts to add the reply to the static buffer in the client struct.
- * Returns C_ERR if the buffer is full, or the reply list is not empty,
- * in which case the reply must be added to the reply list. */
-int _addReplyToBuffer(client *c, const char *s, size_t len) {
-    size_t available = sizeof(c->buf)-c->bufpos;
+ * Returns the lengh of data that is added to the reply list. */
+size_t _addReplyToBuffer(client *c, const char *s, size_t len) {
```

Comment:
Note that this change: always using some of `c->buf`, can potentially more `write` system calls. In the past trying to write a 100k output would never split, but now it will and a small portion (16k) will be written with one `write` call and the rest with another. This can have performance implications.
@oranagra suggested moving this to a seperate PR since it's not directly related, and have the discussion whether it's a good idea or not there.

Reply:
Wow, i am sure what you said is right, in redis protocol, we need to send `*\r\n`,`$\r\n` firstly even if we reply a big bulk. Initially, i want to make full use of `buf`, so i did that. 
need i open a new PR for this change( change 3 as you said above) @oranagra

Reply:
following up on this topic in this thread (responding to later comments)
if we want, we can try to resolve that concern and have an efficient code in both scenarios.
if the `buf` is completely empty (and maybe also the new payload to add is bigger than `PROTO_REPLY_CHUNK_BYTES`), we create a new node, and leave `buf` completely unused. but if `buf` is not empty (it'll have a `write` call anyway), we can utilize it's space.

i'm not certain this complication is worth it, although that's probably just 2 lines of code, and another 2 lines of comment.

Reply:
@ShooterIT do you wanna have a go at my above suggestion?
if not, do we have any other pending issues or can this be merged?
i may have lost track of what's left to do
@yoav-steinberg ^^

Reply:
@ShooterIT convinced my that there's no point in having such an optimization (In implementing your suggestion). This is because `buf` will never be empty, there's always someone that already wrote some '$' or '*' into it. So we're better off using whatever space is available. I'm approving the current PR.